audio-worklet/src/bitcrusher-demo.js

```diff
@@ -16,24 +16,52 @@
 class BitcrusherDemo {
   /**
    * Initalizes and modifies bitcrusher settings in response to GUI input.
-   * @param  {Number} bitDepth the number of bits for each sample
-   * @param  {Number} reduction the amount of sample rate reduction to apply
-   * @param  {Number} gain the volume of the output
+   * @param {AudioContext} context the Audio Context
+   * @param {String} containerId container for GUI elements
    */
-  constructor(containerId, bitDepth, reduction, gain) {
-    this.context_ = new AudioContext();
-    this.masterGain_ = new GainNode(this.context_, {gain: (gain || 0.5)});
-    this.bitcrusher_ = new Bitcrusher(this.context_, {
-      channelCount: 1,
-      bitDepth: bitDepth || 24,
-      reduction: reduction || 1
-    });
+  constructor(context, containerId) {
+    this.context_ = context;
+    this.masterGain_ = new GainNode(this.context_, {gain: 0.5});
+
+    // Create a bit crusher with the audio worklet node.
+    this.bitcrusherAudioWorklet_ =
+        new AudioWorkletNode(this.context_, 'bitcrusher-audio-worklet');
+    this.paramBitDepth_ =
+        this.bitcrusherAudioWorklet_.parameters.get('bitDepth');
+    this.paramReduction_ =
+        this.bitcrusherAudioWorklet_.parameters.get('reduction');
+
+    // GUI default settings depend on parameter values defined in audio-worklet.
+    this.bitDepthMaxValue_ = this.paramBitDepth_.maxValue;
+    this.bitDepthMinValue_ = this.paramBitDepth_.minValue;
+    this.bitDepthDefaultValue_ = this.paramBitDepth_.defaultValue;
+    this.reductionMaxValue_ = this.paramReduction_.maxValue;
+    this.reductionMinValue_ = this.paramReduction_.minValue;
+    this.reductionDefaultValue_ = this.paramReduction_.defaultValue;
+
+    // For the audio worklet, the bit depth and reduction parameters are
+    // modulated by changing the offset to connected constant source nodes.
```

Comment:
My mistake. I thought that since the params needed to be driven by a constant source node they also needed to be modified by that constant source node, but that is not the case.